apps/src/aichat/types.ts

```diff
@@ -146,3 +146,37 @@ export interface LevelAichatSettings {
 
 // The type of save action being performed (customization update, publish, model card save, etc).
 export type SaveType = 'updateChatbot' | 'publishModelCard' | 'saveModelCard';
+
+/** Response structure for the detect toxicity API */
+export interface DetectToxicityResponse {
+  flaggedFields: FlaggedField[];
+}
+
+export interface FlaggedField {
+  field: keyof AiCustomizations;
```

Comment:
`field` can only be `systemPrompt` |  `retrievalContexts`.  Maybe be explicit about this?

Reply:
Good call - created a separate constant `FIELDS_CHECKED_FOR_TOXICITY` to automatically create a type for this, that we can also update later on if we want to check other fields at any point.